Pull Request #1: ADD dynamic search for blog pages

AaronBinChen opened the pull request.

AaronBinChen committed:
1. Modified gatsby-config.js to attempt to convert markdown pages into webpages with GraphQL and Gatsby
2. Modified src/components/Community/index.js - changed line23 from w-max to w-fit. This allows for responsive design to occur on mobile devices
3. Modified src/components/Header/Nav.jsx - changed className of flex to w-fit flex. This allows for responsive design to occur on mobile devices. In addition, I linked two more pages called user stories and one called MarkDown Pages. The latter was done to attempt to convert markdown pages into webpages.
4. Modified src/components/Header/index.jsx - Cjamged ;ome 8 classnMae from w-max to lg:w-max md:w-fit and sm:w-min. Allows for responsive ddesign to occur.
5. Modified src/components/Home/CommunityShowcase.jsx - Line 49 was changed to sm:col-span-12 and md:col-span-6 to allow for easier readability on mobile screens
6. Modified src/components/Layout/index.jsx - changed css of const root to max-w-max and const content to max-w-fit. Added few comments to keep track of where things are located.
7. Modified src/,markdown-pages/post1.md - markdown file of Anubis guide. Attempting to convert to webpage
8. Modified src/pages/blog.js - responsive design
9. New src/pages/userstories.js

meishinlee committed:
ADD markdownfile /first-post

dzhu7400 committed:
FIX markdown page images not showing up

meishinlee committed:
create a function to show all blogs

meishinlee committed:
ADD get all blogs function

meishinlee committed:
ADD blogFilter function to filter blog titles by search term match

meishinlee committed:
Merge branch 'master' of https://github.com/AaronBinChen/AnubisWebsite

meishinlee committed:
ADD getBlogFromBlogFolder function which will go through the directory and get the returns titles into a list

AaronBinChen committed:
Created two new files: blog-post1.md and blog-post2.md. Modified index.md to be our Index file that host all blog posts. Changed post.js to exclude frontmatter title.

AaronBinChen committed:
Making html/css changes to blog-post1.md so that the Anubis markdown file shows up correctly

meishinlee committed:
ADD filter js work into blogs pages/blog.js, display blog return results to the screen

dzhu7400 committed:
Testing the filter function with two blog posts that were added

AaronBinChen committed:
Added search bar to blog.js. Added different styling to global.css for easier visibility while debugging

Co-authored by: Aaron Chen aaron.chen11219@gmail.com
Co-authored by: Mei Shin Lee msl608@nyu.edu
Co-authored by: Devin Zhu dz1295@nyu.edu

AaronBinChen committed:
Reorganized file structure

meishinlee committed:
UPDATE search bar to be dynamic

meishinlee committed:
Merge branch 'master' of https://github.com/AaronBinChen/AnubisWebsite

AaronBinChen committed:
REMOVED button. ADDED dynamic search bar.

Co-Authored By: Mei Shin Lee
Co-Authored By: Devin Zhu
Co-Authored By: Aaron Chen

meishinlee committed:
FIX bug where there are no blogs found

dzhu7400 committed:
ADD dummy data to test search bar

dzhu7400 committed:
ADD JSON dummy data to test search bar on main blog page

Co-authored by Aaron, Mei, and Devin

synoet commented on src/pages/test_data.json:
```diff
@@ -0,0 +1,90 @@
+
+
+
+]
```
note: make sure your remove this before merging

AaronBinChen replied:
Removed

synoet commented on src/pages/userstories.js:
```diff
@@ -0,0 +1,27 @@
+          <p className='text-xl text-gray'>A collection of stories and experiences by our users.</p>
+        </div>
+        <div className='flex flex-col items-center'>
+          <div/>
```
take this out

AaronBinChen replied:
Removed

synoet commented on src/templates/test.js:
```diff
@@ -0,0 +1,18 @@
+    return filteredBlogTitles;
+}
+
+console.log(filterBlogs("Blog", blogTitles));
```
note: remove before merging

AaronBinChen replied:
Removed.

synoet commented on gatsby-node.js:
```diff
@@ -0,0 +1,35 @@
+    })
+
+  })
+}
```
sync with other team, on this.

synoet commented on package.json:
```diff
@@ -26,7 +26,9 @@
     "gatsby-source-filesystem": "^3.14.0",
+    "gatsby-transformer-remark": "^5.6.0",
     "gatsby-transformer-sharp": "^3.14.0",
+    "list-react-files": "^0.2.0",
```
dont need this

synoet commented on src/pages/blog.js:
```diff
@@ -1,20 +1,60 @@
+
+  const [searchInput, setSearchInput] = useState('');
+  var filteredBlogsList = filterBlogs(searchInput, blogTitles);
+  let filteredBlogs = filteredBlogsList.map((item,index)=>{
+    return <li key={index}>{item}</li>
+  })
```
You can just move this back into the body below.

synoet commented on src/pages/blog.js:
```diff
@@ -1,20 +1,60 @@
           <p className='text-xl text-gray'>A collection of posts by our developers and maintainers.</p>
+          <div class="mb-3 xl:w-96">
+          <div class="input-group relative flex flex-wrap items-stretch w-full mb-4">
+              <input type="Search for Blog" onChange = {(e) => { 
+                searchItems(e.target.value); }}
```
replace with `onChange = {(e) => setSearchInput(event.target.value)}`

synoet commented on src/pages/blog.js:
```diff
@@ -1,20 +1,60 @@
   }
+
+  const [searchInput, setSearchInput] = useState('');
+  var filteredBlogsList = filterBlogs(searchInput, blogTitles);
```
You need to store the current blog posts selected in state.

1. You need some state variable representing the state of the blogs list.
`const [filteredBlogs, setFilteredBlogs] = useState(<<originalBlogList>>)`

2. You need to dynamically update the state depending on whether the search input state has changed.
`useEffect(() => {
  setFilteredBlogs(filterBlogs(searchInput)) 
}, [searchInput])`
** note: add searchInput as a dependency to useEffect so that it will be called on change.

AaronBinChen replied:
Used another state variable for blogs list but did not need second state for search input state.

AaronBinChen committed:
RM unnecessary directories such as feb9 and markdown-pages
RM unnecessary userstories.js from src/pages
RM unnecessary files in templates folder such as filter.js, post.js and test.js
ADD new directories content/blog/helloworld and content/blog/helloworld2. Integration with Anubis2 Team
CHG gatsby-config.js, gatsby-node.js and blog.js to integrate with Anubis2 Team
CHG src/components/Header/Nav.jsx to remove userstories from the NavBar
Co-authored by Aaron Chen, Devin Zhu, Mei Shin Lee

AaronBinChen committed:
FIX merge errors with blog.js due to not pulling ahead of time

AaronBinChen committed:
CHG blog.js to only filter non-dummy data such as Anubis2's made posts
Coauthored by Aaron Chen, Devin Zhu, Mei Shin Lee

AaronBinChen committed:
RM blog-posts.js and renamed as BlogPost.js
CHG gatsby-node.js to fix routing errors
CHG blog.js and fixed the filtering on non-dummy data
Coauthored by Aaron Chen, Devin Zhu, Mei Shin Lee

AaronBinChen committed:
CHG blog.js by cleaning up unnecessary modules and comments

AaronBinChen committed:
ADD package-lock.json to gitignore
RM test_data.json
